zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/DocumentenApiClient.kt

```diff
@@ -348,8 +348,11 @@ class DocumentenApiClient(
         }
 
         runWithoutAuthorization {
-            check(getInformatieObject(authentication, documentUrl).status != DocumentStatusType.DEFINITIEF) {
-                "InformatieObject ${documentUrl.path.substringAfterLast("/")} with status 'definitief' cannot be updated!"
+            check(
+                patchDocumentRequest.lock != null
```

Comment:
This doesn't seem to match the business rules. I found the following text in the api documentation:

```
De voorwaarde "status NIET definitief" is verwijderd uit de beschrijvingen van de PUT- en PATCH-operatie op enkelvoudiginformatieobjecten. Dit is een correctie van een fout in de OAS, want deze voorwaarde was al geschrapt in de aanvullende specificatie van versie 1.4.0.
```

This leads me to believe that this check only applies to documenten API versions < 1.4

https://vng-realisatie.github.io/gemma-zaken/standaard/documenten/redoc-1.5.0